src/analysis/complex/basic.lean

```diff
@@ -267,6 +267,16 @@ lemma isometry_of_real : isometry (coe : ℝ → ℂ) := of_real_li.isometry
 
 @[continuity] lemma continuous_of_real : continuous (coe : ℝ → ℂ) := of_real_li.continuous
 
+/-- The only continuous ring homomorphism from `ℝ` to `ℂ` is the identity. -/
+lemma ring_hom_eq_of_real_of_continuous {f : ℝ →+* ℂ} (h : continuous f) :
+  f = complex.of_real :=
+begin
+  convert rat.dense_embedding_coe_real.dense.equalizer h
+    (by continuity! : continuous complex.of_real) _ using 0,
+  { ext1, simp only [fun_like.coe_fn_eq], },
+  { ext1, simp only [function.comp_app, map_rat_cast], }
+end
```

Comment:
Is there anything that can be reused between this and

https://github.com/leanprover-community/mathlib/blob/8315f58ea7ded4917347d7bad0939f6081e5256b/src/analysis/complex/basic.lean#L240-L249

Reply:
Ah nevermind, I was able to golf away almost all of that proof in #17259

Reply:
Nice! I'll remove `ring_hom_eq_of_real_of_continuous` from this PR then.